src/contrib/cluster/Akka.Cluster.Sharding/ShardedDaemonProcess.cs

```diff
@@ -66,18 +74,55 @@ internal sealed class MessageExtractor : HashCodeMessageExtractor
     {
         public MessageExtractor(int maxNumberOfShards)
             : base(maxNumberOfShards)
-        { }
+        {
+        }
+
+        public override string? EntityId(object message) => (message as ShardingEnvelope)?.EntityId;
+        public override object? EntityMessage(object message) => (message as ShardingEnvelope)?.Message;
 
-        public override string EntityId(object message) => (message as ShardingEnvelope)?.EntityId;
-        public override object EntityMessage(object message) => (message as ShardingEnvelope)?.Message;
-        public override string ShardId(string entityId, object messageHint = null)
+        public override string ShardId(string entityId, object? messageHint = null)
         {
             return entityId;
         }
     }
 
     /// <summary>
-    /// <para>This extension runs a pre set number of actors in a cluster.</para>
+    /// Used to support push-based communication with the <see cref="ShardedDaemonProcess"/> - uses
+    /// round-robin messaging to distribute messages to the actors in the cluster.
+    /// </summary>
+    /// <remarks>
+    /// NOTE: does not use a traditional <see cref="Router"/> to distribute messages because we want to use the
+    /// <see cref="ShardRegion"/> mechanism in order to piggyback off of its reliability and message buffering
+    /// features, which <see cref="Router"/> and <see cref="ActorSelection"/> do not support.
+    /// </remarks>
+    internal sealed class DaemonMessageRouter : UntypedActor
```

Comment:
new class - `DaemonMessageRouter`. Performs round-robin message routing (according to the sending node - we don't care what other nodes who might be sending messages are doing) to all of the individual worker actors.